core/templates/pages/contributor-dashboard-admin-page/contributor-dashboard-tables/contributor-admin-stats-table.component.spec.ts

```diff
@@ -301,14 +302,72 @@ describe('Contributor stats component', () => {
         'lastContributedInDays',
         'role',
       ]);
+      expect(component.getOverallItemCount()).toEqual('1+');
+    }));
+
+    it('should check for upperlimit for pagination from total number of items, as its lower than items per page combined with page number.', fakeAsync(() => {
+      component.statsPageNumber = 1;
+      component.itemsPerPage = 4;
+      fixture.detectChanges();
+      component.allStats = [
+        questionReviewerStats,
+        questionReviewerStats,
+        questionReviewerStats,
+      ];
+      expect(component.getIndexOfLastItemOnPage()).toEqual(3);
+    }));
+
+    it('should check for upperlimit for pagination from items per page and page number, on the first page, as its lower than total number of items', fakeAsync(() => {
+      component.statsPageNumber = 0;
+      component.itemsPerPage = 1;
+      fixture.detectChanges();
+      component.allStats = [
+        questionReviewerStats,
+        questionReviewerStats,
+        questionReviewerStats,
+      ];
+      expect(component.getIndexOfLastItemOnPage()).toEqual(1);
     }));
 
-    it('should check for upperlimit for pagination', fakeAsync(() => {
-      component.dataSource = [];
+    it('should check for upperlimit for pagination from items per page and page number, on the second page, as its lower than total number of items', fakeAsync(() => {
       component.statsPageNumber = 1;
+      component.itemsPerPage = 1;
+      fixture.detectChanges();
+      component.allStats = [
+        questionReviewerStats,
+        questionReviewerStats,
+        questionReviewerStats,
+      ];
+      expect(component.getIndexOfLastItemOnPage()).toEqual(2);
+    }));
```

Comment:
I think one of these tests are a duplicate of the other. They are both calculating the index of the last item on the table's page as equal to `statsPageNumber * itemsPerPage + itemsPerPage`. One of these tests should be dropped.

Another telltale sign of this is from the test name. The only difference in the test name is that one occurs on `page 1` and the other on `page 2`. Yet both test names don't indicate that the page number has any significant difference in terms of how the index of the last item on page is calculated.

Reply:
Done